File: Foundation/Android/cn/vimfung/luascriptcore/modules/foundation/Thread.java
package cn.vimfung.luascriptcore.modules.foundation;

import cn.vimfung.luascriptcore.LuaExportType;
import cn.vimfung.luascriptcore.LuaFunction;
import cn.vimfung.luascriptcore.LuaScriptController;
import cn.vimfung.luascriptcore.LuaValue;

/**
 * 线程
 */
public final class Thread implements LuaExportType
{
    /**
     * 线程处理器
     */
    private LuaFunction _handler;

    /**
     * 脚本控制器
     */
    private LuaScriptController _scriptController;

    /**
     * 初始化线程
     * @param handler 线程处理器
     */
    public Thread(LuaFunction handler)
    {
        _scriptController = LuaScriptController.create();
        _handler = handler;
    }

    /**
     * 执行线程
     * @param arguments 参数列表
     */
    public void run(LuaValue[] arguments)
    {
        if (_handler != null)
        {
            _handler.getContext().runThread(_handler, arguments, _scriptController);
        }
    }

    /**
     * 退出线程
     */
    public void exit()
    {
        _scriptController.forceExit();
    }
}

File: Foundation/Unity3D/Thread.cs
using System.Collections;
using System.Collections.Generic;
using UnityEngine;

namespace cn.vimfung.luascriptcore.modules.foundation
{
	/// <summary>
	/// 线程
	/// </summary>
	public class Thread : object, LuaExportType
	{
		private LuaFunction _handler;
		private LuaScriptController _scriptController;

		/// <summary>
		/// 初始化
		/// </summary>
		/// <param name="handler">线程处理器.</param>
		public Thread (LuaFunction handler)
		{
			_scriptController = new LuaScriptController ();
			_handler = handler;
		}

		/// <summary>
		/// 执行线程
		/// </summary>
		/// <param name="arguments">线程参数.</param>
		public void run(List<LuaValue> arguments)
		{
			_handler.context.runThread (_handler, arguments, _scriptController);
		}

		/// <summary>
		/// 退出线程
		/// </summary>
		public void exit()
		{
			_scriptController.forceExit ();
		}
	}
}